### PR TITLE
Add the Update 6 blog

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the Update 6 blog.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]
I added comments to the original google doc. Can you please take a look?

**Comment by praneesha** [comment]
> I added comments to the original google doc. Can you please take a look?

Addressed the comments in this PR.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+- **Persist client API:** This provides an interface to access and manipulate the data stored in your chosen data store. Whether you need to fetch data, update records, or perform other data-related operations, the persist client API simplifies the process.
+
+    The example code below demonstrates how CRUD operations can be performed on the persist model via the client API.       
+
+    ```ballerina
+
+    import ballerina/io;
+    import rainier.store;
+    import ballerina/persist;
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:49]:
```suggestion
    import ballerina/io;
    import ballerina/persist;
    import rainier.store;
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    store:Client sClient = check new ();
+
+    public function main() returns error? {
+        store:EmployeeInsert employee1 = {
+            id: uuid:16c6553a-373c-4b29-b1c8-c282f444248c,
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:55]:
```suggestion
            id: "EMP001",
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            birthDate: {
+                year: 1987,
+                month: 7,
+                day: 23
+            },
+            hireDate: {
+                year: 2020,
+                month: 10,
+                day: 10
+            }
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:68]:
```suggestion
        birthDate: {year: 1987, month: 7, day: 23},
        hireDate: {year: 2020, month: 10, day: 10}
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 2 above)
Comment (daneshk) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md]: Change all places

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        };
+
+    // Create a new employee record.
+    string[] employeeIds = check sClient->/employees.post([employee1]);
+    io:println(string `Inserted employee ids: ${employeeIds.toString()}`);
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:65]:
```suggestion
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    Employee|error updated = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.put({salary: 4000.0});
+
+
+    // Delete the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error deleted = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.delete();
+
+
+    // Get all the employee records.
+    stream<Employee, error?> employee = sClient->/employee;
+    }
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:82]: Not formatted correctly

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    string[] employeeIds = check sClient->/employees.post([employee1]);
+    io:println(string `Inserted employee ids: ${employeeIds.toString()}`);
+
+
+    // Get the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error employee = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c;
+
+
+    // Update the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error updated = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.put({salary: 4000.0});
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:72]:
```suggestion
    store:Employee employee = check sClient->/employees/[empId].put(
        {hireDate: {year: 2014, month: 5, day: 1}}
    );
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        };
+
+    // Create a new employee record.
+    string[] employeeIds = check sClient->/employees.post([employee1]);
+    io:println(string `Inserted employee ids: ${employeeIds.toString()}`);
+
+
+    // Get the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error employee = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c;
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:68]:
```suggestion
    store:Employee|error employee = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c;
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    // Update the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error updated = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.put({salary: 4000.0});
+
+
+    // Delete the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+    Employee|error deleted = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.delete();
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:76]:
```suggestion
    store:Employee|error deleted = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.delete();
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+
+    // Get all the employee records.
+    stream<Employee, error?> employee = sClient->/employee;
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:80]:
```suggestion
    stream<store:Employee, error?> employee = sClient->/employee;
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 6 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md]: Fixed.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 9 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md]: Fixed.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 8 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md]: Fixed.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 5 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:78]: Fixed via https://github.com/ballerina-platform/blog/pull/189#discussion_r1212638058.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        // Get the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+        Employee|error employee = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c;
+
+        // Update the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+        store:Employee employee = check sClient->/employees/[empId].put(
+        {hireDate: {year: 2014, month: 5, day: 1}}
+        );
+
+        // Delete the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
+         store:Employee|error deleted = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.delete();
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:76]:
```suggestion
        // Delete the employee record with the ID: 16c6553a-373c-4b29-b1c8-c282f444248c.
        store:Employee|error deleted = sClient->/employee/16c6553a-373c-4b29-b1c8-c282f444248c.delete();
```

Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:76]:
```suggestion
        // Get the employee record with the ID: EMP001.
        Employee|error employee = sClient->/employee/EMP001;

        // Update the employee record with the ID: EMP001.
        store:Employee employee = check sClient->/employees/[empId].put(
        {hireDate: {year: 2014, month: 5, day: 1}}
        );

        // Delete the employee record with the ID: EMP001.
        store:Employee|error deleted = sClient->/employee/EMP001.delete();
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        string gender;
+        time:Date hireDate;
+    |};
+    ``` 
+
+- **Persist CLI:** This generates a client API for your defined data model. 
+
+- **Persist client API:** This provides an interface to access and manipulate the data stored in your chosen data store. Whether you need to fetch data, update records, or perform other data-related operations, the persist client API simplifies the process.
+
+    The example code below demonstrates how CRUD operations can be performed on the persist model via the client API.
```
Comment (daneshk) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:43]:
```suggestion
    The example code below demonstrates how CRUD operations can be performed on the above model via the client API.       
```

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+<style>.cBlogContent p{white-space: break-spaces !important;}</style>
+
+We are excited to announce the [Ballerina 2201.6.0 (Swan Lake Update 6)](https://ballerina.io/downloads/) release. This latest update brings a host of features and improvements to the Ballerina programming language. 
+
+Out of those, the `Bal persist` feature enhances data storage and retrieval whereas, the other improvements, such as the improved OpenAPI compatibility and the newly introduced Electronic Data Interchange (EDI) support and GraphQL interceptor configurations, further enrich the Ballerina development experience.
+
+**`Bal persist` feature**
+
+One of the major highlights of Ballerina Swan Lake Update 6 is the introduction of the `Bal persist` feature. With this feature, you can now store and retrieve data from various data stores, including in-memory tables, MySQL databases, and Google Sheets. The `Bal persist` feature comprises three primary components:
```
Comment (anupama-pathirage) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:19]: This should be `bal persist`? @daneshk pls confirm?

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+status: Published
+socialmediaimage: Ballerina-Swan-Lake-GA-Release-banner-02-with-button.png
+permalink: /posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6/
+---
+
+<style>.cBlogContent p{white-space: break-spaces !important;}</style>
+
+We are excited to announce the [Ballerina 2201.6.0 (Swan Lake Update 6)](https://ballerina.io/downloads/) release. This latest update brings a host of features and improvements to the Ballerina programming language. 
+
+Out of those, the `Bal persist` feature enhances data storage and retrieval whereas, the other improvements, such as the improved OpenAPI compatibility and the newly introduced Electronic Data Interchange (EDI) support and GraphQL interceptor configurations, further enrich the Ballerina development experience.
```
Comment (anupama-pathirage) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:15]: Do we need to list them down here?

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    ```
+
+- **OpenAPI enum support:** Ballerina Open API tool now supports OpenAPI enums in both Ballerina client and service generation. The tool will generate Ballerina union types to represent the counterpart OpenAPI enum types.
+
+- **GraphQL field interceptors and interceptor configurations:** Starting from this release, you can utilize GraphQL field interceptors to apply an interceptor specifically to a particular resolver (resource/remote) method. Also, the newly introduced interceptor configurations allow you to change the behavior of a GraphQL service interceptor.
+
+For a comprehensive overview of all the changes and improvements brought by Swan Lake Update 6, see the [release note](https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.6.0).
+
+We extend our sincere gratitude to the Ballerina community for their invaluable feedback and continuous support. It is through the community involvement that Ballerina continues to evolve and thrive. We invite you to explore Swan Lake Update 6, explore the new features, and actively contribute to shaping the future of the Ballerina language.
```
Comment (anupama-pathirage) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:110]: IMO this is too lengthy.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+        foreach sorder:Items_Type item in simpleOrder.items {
+            io:println("Item: ", item.item, ", Quantity: ", item.quantity);
+        }
+    }
+    ```
+
+- **OpenAPI enum support:** Ballerina Open API tool now supports OpenAPI enums in both Ballerina client and service generation. The tool will generate Ballerina union types to represent the counterpart OpenAPI enum types.
+
+- **GraphQL field interceptors and interceptor configurations:** Starting from this release, you can utilize GraphQL field interceptors to apply an interceptor specifically to a particular resolver (resource/remote) method. Also, the newly introduced interceptor configurations allow you to change the behavior of a GraphQL service interceptor.
```
Comment (anupama-pathirage) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:106]: `Starting from this release` is not needed?

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md [hunk 16]
```diff
@@ -0,0 +1,112 @@
...
+        store:Employee|error deleted = sClient->/employee/EMP001.delete();
+
+        // Get all the employee records.
+        stream<store:Employee, error?> employee = sClient->/employee;
+    }
+    ```
+
+In addition to the `Bal persist` feature, Swan Lake Update 6 brings a range of notable additions and improvements as described below.
+
+- **Electronic Data Interchange (EDI) support:** Swan Lake Update 6 introduces EDI message processing support in Ballerina. This capability allows the conversion of EDI messages to Ballerina records and vice versa, leveraging a specified EDI schema. Furthermore, it enables the generation of Ballerina records for EDI schemas in handling EDI data.
```
Comment (anupama-pathirage) [inline _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:83]: Do we need the last sentence as the previous one said `vice versa`?

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 13 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md]: Removed.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 15 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:104]: Changed to `now`.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 16 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:83]: Removed.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 14 above)
Comment (praneesha) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:108]: Updated to `We thank the Ballerina community for your invaluable support and feedback in shaping it into what it is today. We are excited to present this release and hope you enjoy it.`.

File: _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md (hunk 12 above)
Comment (daneshk) [reply _posts/2023-05-31-announcing-ballerina-2201.6.0-swan-lake-update-6.md:17]: Yes, it is `bal persist`

**Review by sameerajayasoma — APPROVED** [review]

**Pull request merged**